pkg/psmdb/getters.go

```diff
@@ -67,7 +68,24 @@ func GetRSPods(ctx context.Context, k8sclient client.Client, cr *api.PerconaServ
 			return rsPods, errors.Wrap(err, "failed to list pods")
 		}
 
-		rsPods.Items = append(rsPods.Items, compPods.Items...)
+		rs := cr.Spec.Replset(rsName)
+		var podsLen int
+		switch sts.Labels["app.kubernetes.io/component"] {
+		case "arbiter":
+			podsLen = int(rs.Arbiter.Size)
+		case "nonVoting":
+			podsLen = int(rs.NonVoting.Size)
+		default:
+			podsLen = int(rs.Size)
+		}
+		sort.Slice(pods.Items, func(i, j int) bool {
+			return pods.Items[i].Name < pods.Items[j].Name
+		})
+		if len(pods.Items) >= int(podsLen) {
+			pods.Items = pods.Items[:podsLen]
+		}
+
+		rsPods.Items = append(rsPods.Items, pods.Items...)
```

Comment:
@pooknull , could you please just briefly describe why this logic is needed, I can't understand the reasoning behind it.